### PR TITLE
[G-API] Pipeline modeling tool: Support num_iters criteria

### PR DESCRIPTION
**Pull request by TolyaTalamanov**
### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [ ] I agree to contribute to the project under Apache 2 License.
- [ ] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [ ] The PR is proposed to the proper branch
- [ ] There is a reference to the original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

### Motiviation

Let's image use case where execution should last: 50ms

```
0ms       10ms      20ms      30ms        40ms       50ms
 |---------|---------|----------|----------|----------|
```

and one pipeline iterations takes 15ms.

In that case the picture will be:
```
       0 frame          1 frame       2 frame         3 frame
 |----------------|---------------|---------------|---------------|

0ms       10ms      20ms      30ms        40ms         50ms       60ms
 |**********|**********|**********|**********|**********|**********|
 
 FPS = 4 frames * 1000 / 60ms = 66.6 FPS 
```

But if user enabled `--drop_frames` the picture would be different:

```
       0 frame               1 frame              2 frame
 |----------------|    |---------------|      |---------------|

0ms       10ms        20ms      30ms        40ms         50ms  55ms
 |**********|*****-----|**********|*****------|**********|*****|
 
 FPS = 3 frames * 1000 / 55ms = 54.54 FPS 
```

So `--drop_frames` could significantly affect `FPS`, because number of iterations depends on this flag.
But what if user want to run certain number of iteration but at the same time schedule pipelines with `--drop_frames` mode?

### Implementation

Now every `Pipeline` could have local `num_iters` property. The logic is following:
* `num_iters` is local property for `Pipeline`. (no global `num_iters`)
* if `num_iters` is set then it takes priority over `work_time`.
* `work_time` is still global (no local `work_time` per `Pipeline`), but now is not necessary to be specified.
* If `Pipeline` doesn't have `num_iters` property and global `work_time` isn't specified then throw error.

### CODE REVIEW

File: modules/gapi/samples/pipeline_modeling_tool.cpp [hunk 1]
```diff
@@ -347,10 +347,14 @@ int main(int argc, char* argv[]) {
                 std::map<std::string, std::string>{{"CACHE_DIR", cached_dir}};
         }
 
-        const double work_time_ms =
-            check_and_read<double>(fs, "work_time", "Config");
-        if (work_time_ms < 0) {
-            throw std::logic_error("work_time must be positive");
+        auto opt_work_time_ms = readOpt<double>(fs["work_time"]);
```
Comment (TolyaTalamanov) [inline modules/gapi/samples/pipeline_modeling_tool.cpp:350]: These changes will affect python tests.

File: modules/gapi/samples/pipeline_modeling_tool/pipeline.hpp [hunk 2]
```diff
@@ -40,35 +40,45 @@ std::string PerfReport::toStr(bool expand) const {
     return ss.str();
 }
 
+class StopCriteria {
```
Comment (TolyaTalamanov) [inline modules/gapi/samples/pipeline_modeling_tool/pipeline.hpp:43]: Probably over design. 
On the other hand what if another `stop criteria` will be needed (e.g N iters, but no more than Xms).

File: modules/gapi/samples/pipeline_modeling_tool.cpp (hunk 1 above)
Comment (TolyaTalamanov) [reply modules/gapi/samples/pipeline_modeling_tool.cpp:350]: Done

**Comment by TolyaTalamanov** [comment]
@smirnov-alexey @dmatveev Could you have a look, please?

File: modules/gapi/samples/pipeline_modeling_tool/pipeline.hpp (hunk 2 above)
Comment (dmatveev) [reply modules/gapi/samples/pipeline_modeling_tool/pipeline.hpp:43]: I believe it is Criterion, not Criteria (plural form)

**Review by dmatveev — APPROVED** [review]
:+1:

File: modules/gapi/samples/pipeline_modeling_tool/pipeline.hpp (hunk 2 above)
Comment (TolyaTalamanov) [reply modules/gapi/samples/pipeline_modeling_tool/pipeline.hpp]: Done

**Comment by TolyaTalamanov** [comment]
@asmorkalov Can it be merged?

**Pull request merged**